lib/sync/index.js

```diff
@@ -58,119 +12,16 @@ function toJSON(dataset_id, returnData, cb) {
-    return callback('invalid fn parameter provided in params "' + fn + '"', null);
-  }
-  var fnHandler = module.exports[fn];
-
-  start(function(err) {
-    if (err) {
-      return callback(err);
-    }
-    return fnHandler(dataset_id, params, callback);
-  });
-}
-
-function setConfig(conf) {
-  syncConfig = _.extend({}, DEFAULT_SYNC_CONF, conf);
-}
-
-// Initialise cloud data sync service for specified dataset.
-function init(dataset_id, options, cb) {
-  syncUtil.doLog(dataset_id, 'info', 'init sync for dataset ' + dataset_id + ' with options ' + util.inspect(options));
-  datasets.init(dataset_id, options);
-  start(function(err){
-    if (err) {
-      return cb(err);
-    }
-    syncStorage.updateManyDatasetClients({datasetId: dataset_id}, {stopped: false}, cb);
-  });
-}
-
-// Stop cloud data sync for the specified dataset_id.
-function stop(dataset_id, cb) {
-  if (!syncStarted) {
-    return cb();
-  }
-  syncUtil.doLog(dataset_id, 'info', 'stop sync for dataset');
-  syncStorage.updateManyDatasetClients({datasetId: dataset_id}, {stopped: true}, cb);
-}
-
-// Stop cloud data sync service for ALL datasets and reset.
-// This should really only used by tests.
-function stopAll(cb) {
-  //sync is not started yet, but connect could be called already. In this case, just reset a few things
-  if (!syncStarted) {
-    mongoDbClient = null;
-    redisClient = null;
-    metricsClient = null;
-    return cb();
-  }
-  syncUtil.doLog(syncUtil.SYNC_LOGGER, 'info', 'stopAll syncs');
-  ackQueue.stopPruneJob();
-  pendingQueue.stopPruneJob();
-  syncQueue.stopPruneJob();
-  async.parallel([
-    async.apply(syncStorage.updateManyDatasetClients, {}, {stopped: true}),
-    async.apply(syncWorker.stop.bind(syncWorker)),
-    async.apply(ackWorker.stop.bind(ackWorker)),
-    async.apply(pendingWorker.stop.bind(pendingWorker)),
-    async.apply(syncScheduler.stop.bind(syncScheduler))
-  ], function(err){
-    if (err) {
-      syncUtil.doLog(syncUtil.SYNC_LOGGER, 'error', 'Failed to stop sync due to error : ' + util.inspect(err));
-      return cb(err);
-    }
-    setConfig();
-    dataHandlers.restore();
-    interceptors.restore();
-    mongoDbClient = null;
-    redisClient = null;
-    metricsClient = null;
-    ackQueue = null;
-    pendingQueue = null;
-    syncQueue = null;
-    ackWorker = null;
-    pendingWorker = null;
-    syncWorker = null;
-    syncStarted = false;
-    dataHandlers = null;
-    interceptors = null;
-    hashProvider = null;
-    syncLock = null;
-    return cb();
-  });
-}
-
 function listCollisions(dataset_id, params, cb) {
   syncUtil.doLog(dataset_id, 'info', 'listCollisions');
-  return dataHandlers.listCollisions(dataset_id, params.meta_data, cb);
+  return server.api.listCollisions(dataset_id, params.meta_data, cb);
 }
 
 // Defines a handler function for deleting a collision from the collisions list.
 // Should be called after the dataset is initialised.
 function removeCollision(dataset_id, params, cb) {
```

Comment:
do we still need those 2 functions here? Perhaps we can remove them and just call the ones in `server.api`?